integration-cli/docker_cli_exec_test.go

```diff
@@ -174,29 +143,25 @@ func (s *DockerSuite) TestExecTtyCloseStdin(c *check.C) {
 
 	cmd := exec.Command(dockerBinary, "exec", "-i", "exec_tty_stdin", "cat")
 	stdinRw, err := cmd.StdinPipe()
-	if err != nil {
-		c.Fatal(err)
-	}
+	c.Assert(err, checker.IsNil)
 
 	stdinRw.Write([]byte("test"))
 	stdinRw.Close()
 
-	if out, _, err := runCommandWithOutput(cmd); err != nil {
-		c.Fatal(out, err)
-	}
+	out, _, err := runCommandWithOutput(cmd)
+	c.Assert(err, checker.IsNil, check.Commentf(out))
 
-	out, _ := dockerCmd(c, "top", "exec_tty_stdin")
+	out, _ = dockerCmd(c, "top", "exec_tty_stdin")
 	outArr := strings.Split(out, "\n")
-	if len(outArr) > 3 || strings.Contains(out, "nsenter-exec") {
-		c.Fatalf("exec process left running\n\t %s", out)
-	}
+	c.Assert(len(outArr) > 3, checker.False)
```

Comment:
Helps to reveal intent in cases of failure, also easier to read the code.

As is the failure will read "false != true" rather than giving the actual values.